diesel/src/pg/connection/stmt/mod.rs

```diff
@@ -41,16 +43,34 @@ impl Statement {
             .len()
             .try_into()
             .map_err(|e| crate::result::Error::SerializationError(Box::new(e)))?;
-        unsafe {
-            raw_connection.send_query_prepared(
-                self.name.as_ptr(),
-                param_count,
-                params_pointer.as_ptr(),
-                param_lengths.as_ptr(),
-                self.param_formats.as_ptr(),
-                1,
-            )
-        }?;
+
+        if let Some(ref sql) = self.sql {
+            // For regular unnamed statements, use send_query_params
```

Comment:
It would be great to include the reasons why we should use `send_query_params` here